SnakeYAML has deprecated SafeConstructor()

## Issue
As part of my publish automation work, I enabled javac linting and noticed that the `SafeConstructor()` has been deprecated:

```
[ TASK compile-java  ]-----------------------------------------------------------
compile-java with java version 11.0.16.1
./src/java/clj_yaml/MarkedConstructor.java:24: warning: [deprecation] SafeConstructor() in org.yaml.snakeyaml.constructor.SafeConstructor has been deprecated
        super();
        ^
./src/java/clj_yaml/UnknownTagsConstructor.java:12: warning: [deprecation] SafeConstructor() in org.yaml.snakeyaml.constructor.SafeConstructor has been deprecated
    public UnknownTagsConstructor() {
                                    ^
```

Not sure why this is deprecated, but [the change is recent](https://github.com/snakeyaml/snakeyaml/commit/bece92a6094fea702ab94c1dce2dadfd764ae32b).

## Next Steps
We could live with this, but why would we? It seems like an easy fix.

--- a/CHANGELOG.adoc
+++ b/CHANGELOG.adoc
@@ -15,7 +15,9 @@ Clj-yaml makes use of SnakeYAML, please also refer to the https://bitbucket.org/
 //   (adjust these in publish.clj as you see fit)
 == Unreleased
 
-...
+* Stop using deprecated SnakeYAML constructor
+(https://github.com/clj-commons/clj-yaml/issues/61[#61])
+(https://github.com/lread[@lread])
 
 == v0.7.169 - 2022-09-26
 

--- a/src/java/clj_yaml/MarkedConstructor.java
+++ b/src/java/clj_yaml/MarkedConstructor.java
@@ -1,5 +1,6 @@
 package clj_yaml;
 
+import org.yaml.snakeyaml.LoaderOptions;
 import org.yaml.snakeyaml.constructor.Construct;
 import org.yaml.snakeyaml.constructor.Constructor;
 import org.yaml.snakeyaml.constructor.SafeConstructor;
@@ -21,7 +22,7 @@ public class MarkedConstructor extends SafeConstructor {
     public MarkedConstructor() {
         // Make sure SafeConstructor's constructor is called first,
         // so that we overwrite the keys that SafeConstructor sets.
-        super();
+        super(new LoaderOptions());
         // Wrap all the constructors with Marking constructors.
         for (Tag tag : tags) {
             Construct old = this.yamlConstructors.get(tag);

--- a/src/java/clj_yaml/UnknownTagsConstructor.java
+++ b/src/java/clj_yaml/UnknownTagsConstructor.java
@@ -1,5 +1,6 @@
 package clj_yaml;
 
+import org.yaml.snakeyaml.LoaderOptions;
 import org.yaml.snakeyaml.constructor.Construct;
 import org.yaml.snakeyaml.constructor.Constructor;
 import org.yaml.snakeyaml.constructor.SafeConstructor;
@@ -10,6 +11,7 @@ import org.yaml.snakeyaml.nodes.Tag;
 public class UnknownTagsConstructor extends SafeConstructor {
 
     public UnknownTagsConstructor() {
+        super(new LoaderOptions());
         this.yamlMultiConstructors.put("", new UnknownTagConstruct());
     }
 

